### PR TITLE
ENH: More complete raw reading

### PR DESCRIPTION
**Pull request by Eric89GXL**
Fixes / adds a bunch of fields we didn't have for Raw. I also finally got rid of the `read/write_extra_meas_info` functions, because the fields are now read/written manually. I didn't add any explicit tests because the I/O is implicitly tested already.

### CODE REVIEW

File: mne/io/fiff/raw.py [hunk 1]
```diff
@@ -158,6 +158,23 @@ def _preload_data(self, preload):
...
+        verbose : bool, str, int, or None
+            If not None, override default verbose level (see mne.verbose).
+
+        Notes
+        -----
+        This function will preload raw data if it was not already preloaded.
+        If data were already preloaded, it will do nothing.
+        """
+        if not self.preload:
+            self._preload_data(True)
```
Comment (Eric89GXL) [inline mne/io/fiff/raw.py]: I've wanted to be able to do this before. Now we can...

Comment (agramfort) [inline mne/io/fiff/raw.py]: fine with me. I'd document this in API section in what's new though

File: mne/io/constants.py [hunk 2]
```diff
@@ -747,18 +750,7 @@ def __init__(self, **kwargs):
...
-#
-FIFF.FIFFV_POINT_CARDINAL = 1
-FIFF.FIFFV_POINT_HPI      = 2
-FIFF.FIFFV_POINT_EEG      = 3
-FIFF.FIFFV_POINT_ECG      = FIFF.FIFFV_POINT_EEG
-FIFF.FIFFV_POINT_EXTRA    = 4
 
-FIFF.FIFFV_POINT_LPA      = 1
-FIFF.FIFFV_POINT_NASION   = 2
-FIFF.FIFFV_POINT_RPA      = 3
```
Comment (agramfort) [inline mne/io/constants.py]: why removing these?

**Comment by agramfort** [comment]
travis is not happy

that's it for me

File: mne/io/constants.py (hunk 2 above)
Comment (Eric89GXL) [inline mne/io/constants.py]: they are already defined elsewhere in the file

Comment (agramfort) [inline mne/io/constants.py]: ok

**Comment by Eric89GXL** [comment]
Fixed tests, updated `whats_new.py`, and unified some `Info` handling for our non-FIFF formats. Tests pass here and on Travis/AppVeyor. Ready for review/merge from my end.

File: mne/io/meas_info.py [hunk 3]
```diff
@@ -693,11 +781,75 @@ def write_meas_info(fid, info, data_type=None, reset_range=True):
...
+            write_float(fid, FIFF.FIFF_SFREQ, hpi_meas['sfreq'])
+        if hpi_meas.get('nchan') is not None:
+            write_int(fid, FIFF.FIFF_NCHAN, hpi_meas['nchan'])
+        if hpi_meas.get('nave') is not None:
+            write_int(fid, FIFF.FIFF_NAVE, hpi_meas['nave'])
+        if hpi_meas.get('ncoil') is not None:
+            write_int(fid, FIFF.FIFF_HPI_NCOIL, hpi_meas['ncoil'])
+        if hpi_meas.get('first_samp') is not None:
+            write_int(fid, FIFF.FIFF_FIRST_SAMPLE, hpi_meas['first_samp'])
+        if hpi_meas.get('lastmne/viz/tests/test_montage.py_samp') is not None:
```
Comment (agramfort) [inline mne/io/meas_info.py]: why this?

**Comment by agramfort** [comment]
besides LGTM

can somebody else have a look?

File: mne/io/meas_info.py (hunk 3 above)
Comment (Eric89GXL) [inline mne/io/meas_info.py]: Whoa, brutal typo

**Comment by Eric89GXL** [comment]
Typo fixed.

File: doc/source/whats_new.rst [hunk 4]
```diff
@@ -111,6 +111,8 @@ API
 
    - When computing the noise covariance or MNE inverse solutions, the rank is estimated empirically using more sensitive thresholds, which stabilizes results by `Denis Engemann`_ and `Eric Larson`_ and `Alex Gramfort`_
 
+   - Raw FIFF files can be preloaded after class istantiation using ``raw.preload_data()``.
```
Comment (mainakjas) [inline doc/source/whats_new.rst]: instantiation

**Comment by agramfort** [comment]
thanks @Eric89GXL 

I'll fix typo in master

**Pull request merged**